src/test/java/spoon/test/parent/ContractOnSettersParametrizedTest.java

```diff
@@ -49,23 +48,29 @@
 import java.util.List;
 import java.util.Set;
 
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 import static spoon.testing.utils.ModelUtils.createFactory;
 
 /**
  * check that all setters of the metamodel do the right things:
  * - call setParent
  * - trigger a change event
   */
-@RunWith(Parameterized.class)
 public class ContractOnSettersParametrizedTest {
 
 	private static final Factory factory = createFactory();
 	private static final List<CtType<? extends CtElement>> allInstantiableMetamodelInterfaces = SpoonTestHelpers.getAllInstantiableMetamodelInterfaces();
+	private ModelChangeListener changeListener = new ModelChangeListener();
 
-	@Parameterized.Parameters(name = "{0}")
-	public static Collection<Object[]> data() {
-		return createReceiverList();
+	@TestFactory
+	public  Collection<DynamicTest> data() {
+		List<DynamicTest> values = new ArrayList<>();
+		for (CtType<?> t : allInstantiableMetamodelInterfaces) {
+			if (!(CtReference.class.isAssignableFrom(t.getActualClass()))) {
```

Comment:
why do we need this extra condition?

Reply:
I copied the called method from here https://github.com/INRIA/spoon/blob/c8ec87305409fd864ec68ca07a94932c91fced99/src/test/java/spoon/test/parent/ContractOnSettersParametrizedTest.java#L76 and adapted it to create DynamicTests. The method `createReceiverList` is used in other test classes and therefore wasn't easy to change.

Reply:
ok thanks